### ISSUE TITLE
[BUG] b2bua does not add < > brackets in generated To-Headers

### ISSUE THREAD

**Issue by xaled1** [OpenSIPS/opensips#2843]
OpenSIPS 3.3 beta b2bua does not add < > brackets in generated To-Headers. 
If there are uri-parameters (e.g. ;user=phone – which is a fairly common situation) this violates rfc3261 and rfc8217. 
Might it be possible to just always add < > brackets around the To-Header? This is simple and always valid, in any case.

It was already documented in this issue but still present in the latest 3.3 beta:
https://github.com/OpenSIPS/opensips/issues/1210

**Comment by bogdan-iancu** [comment]
@xaled1 , as per #1210, the TO hdr is perfectly valid without the angle brackets, as time as you do not have URI parameters. Is in your case the B2B building a TO hdr with URI params and without the brackets? if so, please post the SIP message as generated by B2B as long as the instructions on how you triggered it.

**Comment by xaled1** [comment]
Incoming To-Header:
To:  <sip:+4943xxxxxxxxxx@some.domain;user=phone>

Outgoing To-Header
To:  sip:+4943xxxxxxxxxx@some.domain;user=phone

Relevant line in opensips.cfg:
b2b_init_request("top hiding","p");

**Comment by xaled1** [comment]
Incoming To-Header:
To:  <sip:+4943xxxxxxxxxx@some.domain;user=phone>

Outgoing To-Header
To:  sip:+4943xxxxxxxxxx@some.domain;user=phone

Relevant line in opensips.cfg:
b2b_init_request("top hiding","p");

**Comment by xaled1** [comment]
`Incoming To-Header:
To:  <sip:+4943xxxxxxxxxx@some.domain;user=phone>

Outgoing To-Header
To:  sip:+4943xxxxxxxxxx@some.domain;user=phone

Relevant line in opensips.cfg:
b2b_init_request("top hiding","p");`

**Comment by xaled1** [comment]
```
Incoming To-Header:
To:  <sip:+4943xxxxxxxxxx@some.domain;user=phone>

Outgoing To-Header
To:  sip:+4943xxxxxxxxxx@some.domain;user=phone

Relevant line in opensips.cfg:
b2b_init_request("top hiding","p");
```

**Comment by xaled1** [comment]
Any plans to fix it?

**Comment by lylepratt** [comment]
This is a probem. Any updates?

**Comment by xaled1** [comment]
Dear OpenSIPS team,
will this get fixed in the new OpenSIPS version?

**Comment by bogdan-iancu** [comment]
Fixed via 9af192693f419751ff3fc704d5286274752171b1 , also backported.

**Comment by bhoeneis** [comment]
This appears to be resolved for Requests only. For Replies (e.g. 407), the square brackets are still missing for the To HF:

- In: `To: <sip:0123456789@example.com>;tag=8e14b31b8709851a3d139fba84110914.5f1bdd99` 
- Out: `To: sip:0123456789@example.com;tag=B2B.256.222.1718881474.159853479`

Server: OpenSIPS (3.4.5 (x86_64/linux))
